doc/design/var_desc.md

```diff
@@ -0,0 +1,122 @@
+## Background
+PaddlePaddle divides the description of neural network computation graph into two stages: compile time and runtime.
+
+PaddlePaddle use proto message to describe compile time graph for
+
+1. Computation graph should be able to be saved to a file.
+1. In distributed training, the graph will be serialized and send to multiple workers.
+
+The computation graph is constructed by Data Node and Operation Node. The concept to represent them is in the table below.
+
+| |compile time|runtime|
+|---|---|---|
+|Data|VarDesc(proto)|Variable(cpp)|
+|Operation|OpDesc(proto)|Operator(cpp)|
+
+
+## Definition of VarDesc
+
+A VarDesc should have a name and value, in PaddlePaddle, the value will always be a tensor. Since we use LoDTensor most of the time. We add a LoDTesnorDesc to represent it.
+
+```proto
+message VarDesc {
+  required string name = 1;
```

Comment:
Do we need to use required field? https://stackoverflow.com/a/31814967/852385

Reply:
good point! proto3 even remove `required`. 

But to be compatible with the [current code](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/framework.proto#L32), I want to use required in this PR and create another pr to change all the proto at once.

Reply:
I see, thanks!